src/librustdoc/clean/utils.rs

```diff
@@ -331,11 +334,21 @@ crate fn print_evaluated_const(tcx: TyCtxt<'_>, def_id: DefId) -> Option<String>
 
 fn format_integer_with_underscore_sep(num: &str) -> String {
     let num_chars: Vec<_> = num.chars().collect();
-    let num_start_index = if num_chars.get(0) == Some(&'-') { 1 } else { 0 };
+
+    let mut num_start_index = if num_chars.get(0) == Some(&'-') { 1 } else { 0 };
+    let chunk_size = if matches!(
+        (num_chars.get(num_start_index), num_chars.get(num_start_index + 1)),
+        (Some(&'0'), Some(&'b' | &'o' | &'x'))
+    ) {
+        num_start_index += 2;
+        4
+    } else {
+        3
+    };
```

Comment:
I think it might be nice if this matched what clippy suggests for octal literals in https://github.com/rust-lang/rust/blob/master/src/tools/clippy/clippy_utils/src/numeric_literal.rs#L18, which is groups of 3 rather than 4. I don't know why 3, but maybe it's because of Unix permissions?